File: fluttter/simple_app/lib/domain/models/semester.dart
import 'package:simple_app/domain/models/enums/semester_type.dart';

class Semester{
  final String code;
  final String year;
  final SemesterType semesterType;

  Semester({
    required this.code,
    required this.year,
    required this.semesterType,
  });

  factory Semester.fromJson(Map<String, dynamic> json) {
    return Semester(
        code: json['Code'],
        year: json['Year'],
        semesterType: SemesterType.values[json['SemesterType']],
    );
  }
}

File: dotnet/FinkiRasporedi/Models/Domain/Semester.cs
using FinkiRasporedi.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace FinkiRasporedi.Models.Domain
{
    public class Semester
    {
        [Key]
        public string Code { get; set; }
        public string? Year { get; set; }

        public SemesterType? SemesterType { get; set; }
    }
}
